### PR TITLE
allow createController usage asynchronously without errors

### PR DESCRIPTION
**Pull request by giftkugel**
Based on https://github.com/rico-projects/rico-angular/pull/6
Solves https://github.com/rico-projects/rico/issues/47

### CODE REVIEW

File: src/lib/rico-angular.service.ts [hunk 1]
```diff
@@ -53,10 +57,41 @@ export class RicoService {
...
+        const controllerProxy = new ControllerProxy(name, this.clientContext);
+        this.addToQueue(controllerProxy, resolve, reject);
+      });
     } else {
       return Promise.reject('Cannot create controller. ClientContext not conntected');
     }
   }
+
+  private addToQueue(controllerProxy: ControllerProxy, resolve: Function, reject: Function) {
+    const promise = new Promise((resolve, reject) => {
```
Comment (madmas) [inline src/lib/rico-angular.service.ts]: TSLint warning: Shadowed name: 'resolve'

File: src/lib/rico-angular.service.ts [hunk 2]
```diff
@@ -53,10 +57,41 @@ export class RicoService {
...
+      this.queue.push(resolve);
+    });
+
+    promise.then(() => {
+      controllerProxy.create().then((controllerProxy) => {
```
Comment (madmas) [inline src/lib/rico-angular.service.ts]: [tslint] Shadowed name: 'controllerProxy' [no-shadowed-variable]
(same for reject)

**Comment by hendrikebbers** [comment]
LGTM :)

**Comment by madmas** [comment]
tests! 🎉🤩

**Pull request merged**